app/stac_api/views.py

```diff
@@ -181,7 +181,8 @@ def get_queryset(self):
 
     def list(self, request, *args, **kwargs):
         queryset = self.filter_queryset(self.get_queryset())
-
+        if not Collection.objects.filter(name=self.kwargs['collection_name']).exists():
+            raise Http404("This collection does not exists.")
```

Comment:
This should be moved on top of the method to avoid doing unnecessary queryset. Also add logger.error() message and I would also add the collection name to both the logger message and Http404 message.